pandas/core/indexes/numeric.py

```diff
@@ -394,10 +394,9 @@ def _convert_scalar_indexer(self, key, kind: str):
 
     @Appender(Index._convert_slice_indexer.__doc__)
     def _convert_slice_indexer(self, key: slice, kind=None):
+        assert kind in ["loc", "getitem", None]
 
-        if kind == "iloc":
-            return super()._convert_slice_indexer(key, kind=kind)
-
+        # We always treat __getitem__ slicing as label-based
```

Comment:
I am not sure this comment is correct? (eg for a series with Int64Index, slicing is positional?)

Reply:
this method is for Float64Index

Reply:
Ah, missed that in the diff. Thanks!